api/controllers/arrive.js

```diff
@@ -79,321 +79,310 @@ module.exports = {
+      var zoneLeftLongitudeDeg = inputs.long;
+      var ZONE_WIDTH_IN_DEG = (1 / sails.config.custom.numZonesPerDegreeSquare);
+      return zoneLeftLongitudeDeg + (ZONE_WIDTH_IN_DEG/2);
+    })();//†
+
+    var relevancyRadius = (()=>{
+      var ZONE_HEIGHT_IN_DEG = (1 / sails.config.custom.numZonesPerDegreeSquare);
+      var ZONE_WIDTH_IN_DEG = (1 / sails.config.custom.numZonesPerDegreeSquare);
+      var zoneHeightKm = 111 * ZONE_HEIGHT_IN_DEG;
+      var zoneWidthKm = Math.abs(Math.cos(zoneCenterLatitudeDeg)) * (111*ZONE_WIDTH_IN_DEG);
+      var zoneEdgeHypotenuse = Math.sqrt(Math.pow(zoneHeightKm,2) + Math.pow(zoneWidthKm,2));
+      sails.log.verbose('zoneWidthKm :: '+zoneWidthKm);
+      sails.log.verbose('zoneHeightKm :: '+zoneHeightKm);
+      sails.log.verbose('Math.pow(zoneHeightKm,2) :: '+Math.pow(zoneHeightKm,2));
+      sails.log.verbose('Math.pow(zoneWidthKm,2) :: '+Math.pow(zoneWidthKm,2));
+      sails.log.verbose('Math.sqrt(Math.pow(zoneHeightKm,2) + Math.pow(zoneWidthKm,2)) :: '+Math.sqrt(Math.pow(zoneHeightKm,2) + Math.pow(zoneWidthKm,2)));
+
+      // This little guy forms something close to a "circle of best fit" in our
+      // rectangular zone.
+      return zoneEdgeHypotenuse;
+    })();//†
+
+    sails.log.verbose('Adjusted zone center coordinates: ('+zoneCenterLatitudeDeg+'° N,'+zoneCenterLongitudeDeg+'°)');
+    sails.log.verbose('Relevancy radius: '+relevancyRadius);
+
+    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+    // FUTURE: To improve zone load time (esp. for the first load of the day for that zone),
+    // use `async.auto` (or something comparable) to fetch the weather and search tweets
+    // simultaneously.
+    // - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+    if (!sails.config.custom.twitterConsumerKey || !sails.config.custom.twitterConsumerSecret) {
+      sails.log.info('Using fake Twitter data...\n(To resolve, finish setting up relevant custom config.)');
+      zone.cachedTweets = [];
+    }
+    else {
+
+      // Use cached tweets, if possible -- as long as they're not too old.
+      let rightNow = Date.now();
+      let twoHoursAgo = rightNow - (1000*60*60*2);
+      let tooStale = twoHoursAgo < zone.lastCachedTweetsAt;
+      let isCorrupted = !tooStale && !zone.cachedTweets;
+      if (tooStale || isCorrupted)  {
+
+        // FUTURE: Probably temporarily cache the bearer token, rather than looking it up every time.
+        // (But note that it expires)
+        let bearerToken = await Twitter.getBearerToken({
+          consumerKey: sails.config.custom.twitterConsumerKey,
+          consumerSecret: sails.config.custom.twitterConsumerSecret,
+        });
+
+        sails.log.verbose('Searching for tweets from ('+inputs.lat+'° N,'+inputs.long+'°)');
+        sails.log.verbose('(Note that adjusted zone center coordinates will be used instead...)');
+
+        let matchingTweets;
+        try {
```

Comment:
@juliandavidmr same deal here -- in this case, we're intercepting all errors though, so it'd just be like:

```
var matchingTweets = await Twitter.searchTweets.with({
  //…
})
.intercept((err)=>{
  return new Error(
    'Unable to get/cache relevant tweets!  (Be sure to check your configuration, and/or the details below.)\n'+
    'Error details:\n'+
    '-- -- -- -- -- -- -- -- --\n'+
    err.stack+'\n'+
    '-- -- -- -- -- -- -- -- --'
  );
});